syntax = "proto3";
package csce662;
import "google/protobuf/empty.proto";
import "sns.proto";

service CoordService{
    rpc Register (ServerRegistration) returns (google.protobuf.Empty);
    rpc Heartbeat (ServerID) returns (google.protobuf.Empty) {}
    rpc GetServer (ClientID) returns (ServerInfo) {}
}

enum ServerCapability{
    SNS = 0;
}

message ServerID{
    int32 id = 1;
}

message ServerRegistration {
    ServerInfo info = 1;
    repeated ServerCapability capabilities = 2; // what the server can do
}

message ServerInfo{
    int32 id = 1;
    string hostname = 2;
    int32 port = 3;
}


